### PR TITLE
Make installable package

### PR DESCRIPTION
**Pull request by igordertigor**
Hi @juanmc2005 , here is an installable package as discussed in #2.

In order to make things work with the python packaging system, had to do the following changes
1. Assign a package name. I picked the rather long `streaming_speaker_diarization`, as that most closely matches the repository name. Feel free to pick something else. Right now, external code needs to `import streaming_speaker_diarization.pipelines` and similar, which is a lot of typing.
2. I had to group your individual modules (i.e. single files) into a package (folder with the files) that can be installed as one. That required introducing a separate folder with the package name. So your source packages are now in `src/streaming_speaker_diarization/`. An alternative setup could have been to replace the `src/` folder by a folder named `streaming_speaker_diarization/`. I felt that keeping `src/` was closer to your original layout.
3. I modified the imports. You had imported each module individually, which doesn't make sense anymore once they are grouped into a package. The imports now explicitly use local imports such as `from . import sources as src`. I import with the full package name only in the `main.py` so that you can still run `main.py` as a script. However, you would need to add the `src` folder to your `PYTHONPATH`. Once you installed the package, you can also run your main script as `python -m streaming_speaker_diarization.main`. I didn't mark the main script as an executable script as that would have probably meant picking a different name than `main` for it and I didn't want to interfere with your choices there.
4. I added the typical setup files (`setup.cfg`, `setup.py` and `pyproject.toml`). You can build a package (python wheel) by running `python -m build --wheel`. By having the `setup.py` you can probably also just pip-install from the git-repository itself. The setup files contain some classifiers. Let me know if those are appropriatse.

Please feel free to make any change that you see fit. If you are busy with other things, I'm also happy to apply any changes that you suggest.

### CODE REVIEW

**Review by juanmc2005 — CHANGES_REQUESTED** [review]
Hey, sorry it took me a while to get to this PR. I've been quite busy with other things.
I also thought a change of name for the package was a must and I wanted to make sure I had an appropriate one.

The PR is looking good! Just some minor changes about naming.

File: src/streaming_speaker_diarization/main.py [hunk 1]
```diff
@@ -1,9 +1,11 @@
-import sources as src
-from pipelines import OnlineSpeakerDiarization
-from sinks import OutputBuilder
 from pathlib import Path
 import argparse
```
[the remark this answers is not in the document]
Comment (igordertigor) [reply src/streaming_speaker_diarization/main.py]: Covered by 901cef7 (and partly also by 3537a2c)

**Comment by igordertigor** [comment]
Thanks for the feedback. If you don't mind, please take a look at what I did and if that meets your expectations.

**Comment by juanmc2005** [comment]
Awesome, thank you for the quick changes. I'll make a test run on my end tomorrow morning and merge if everything goes well

**Review by juanmc2005 — APPROVED** [review]

**Pull request merged**